sonar-kotlin-plugin/src/main/java/org/sonarsource/kotlin/plugin/KotlinSensor.kt

```diff
@@ -127,6 +127,27 @@ class KotlinSensor(
         sensorDuration?.stop()
     }
 
+    private fun reuseCPDTokens(inputFile: InputFile,sensorContext: SensorContext): Boolean {
+        if (!sensorContext.isCacheEnabled) {
+            return false
+        }
+        val cpdTokensCache = CPDTokensCache(sensorContext.previousCache(), sensorContext.nextCache())
+        val previousTokens: List<CPDToken>? = cpdTokensCache.loadFromPreviousAnalysis(inputFile)
+        return if (previousTokens != null) {
+            val tokensToReuse = sensorContext.newCpdTokens().onFile(inputFile)
+            previousTokens.forEach { tokensToReuse.addToken(it.range, it.text) }
+            tokensToReuse.save()
+            try {
+                cpdTokensCache.copyForNextAnalysis(inputFile)
+                true
+            } catch (_: IllegalArgumentException) {
+                false
```

Comment:
We don't use cached tokens for this analysis if we can't cache them for the next analysis? Not saying this is the wrong decision, I'm curious about the background

Reply:
Add a log message to warn of an issue that does not make the analysis fail